### PR TITLE
HHH-15398 Move to Jakarta Persistence 3.1 as baseline

### PR DESCRIPTION
**Pull request by mbladel**
https://hibernate.atlassian.net/browse/HHH-15398

Also cleaned up some compatibility logic for Jakarta 3.0 introduced with https://github.com/hibernate/hibernate-orm/pull/4832 since it's no longer needed.

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/id/factory/internal/StandardIdentifierGeneratorFactory.java [hunk 1]
```diff
@@ -108,11 +108,7 @@ private void registerJpaGenerators() {
 		generatorTypeStrategyMap.put( GenerationType.SEQUENCE, SequenceGenerationTypeStrategy.INSTANCE );
 		generatorTypeStrategyMap.put( GenerationType.TABLE, TableGenerationTypeStrategy.INSTANCE );
 		generatorTypeStrategyMap.put( GenerationType.IDENTITY, IdentityGenerationTypeStrategy.INSTANCE );
-		try {
-			generatorTypeStrategyMap.put( GenerationType.valueOf( "UUID" ), UUIDGenerationTypeStrategy.INSTANCE );
-		}
-		catch (IllegalArgumentException ignore) {
-		}
+		generatorTypeStrategyMap.put( GenerationType.valueOf( "UUID" ), UUIDGenerationTypeStrategy.INSTANCE );
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/id/factory/internal/StandardIdentifierGeneratorFactory.java:111]: If you change the dependency version you should be able to use this:

```suggestion
		generatorTypeStrategyMap.put( GenerationType.UUID, UUIDGenerationTypeStrategy.INSTANCE );
```

File: hibernate-core/src/main/java/org/hibernate/bytecode/spi/ClassTransformer.java [hunk 2]
```diff
@@ -35,5 +37,5 @@ byte[] transform(
 			String className,
 			Class<?> classBeingRedefined,
 			ProtectionDomain protectionDomain,
-			byte[] classfileBuffer);
+			byte[] classfileBuffer)  throws TransformerException;
```
Comment (mbladel) [inline hibernate-core/src/main/java/org/hibernate/bytecode/spi/ClassTransformer.java:40]: This is a change to an SPI method but should be inline with the [Jakarta 3.1 spec](https://jakarta.ee/specifications/persistence/3.1/apidocs/jakarta.persistence/jakarta/persistence/spi/classtransformer) of this interface.

Comment (beikov) [reply hibernate-core/src/main/java/org/hibernate/bytecode/spi/ClassTransformer.java:40]: Yeah, this is fine

File: hibernate-core/src/main/java/org/hibernate/id/factory/internal/StandardIdentifierGeneratorFactory.java (hunk 1 above)
Comment (mbladel) [reply hibernate-core/src/main/java/org/hibernate/id/factory/internal/StandardIdentifierGeneratorFactory.java]: Correct, I missed it - now it's fixed thanks!

**Review by beikov — APPROVED** [review]

**Pull request merged**